### PR TITLE
Invalidating a cookie based session when user has logged out from another browser.

### PR DESCRIPTION
**Pull request by kumards**
Fix for #7410.

### CODE REVIEW

File: src/Orchard.Web/Modules/Orchard.Users/Services/SecurityFilter.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            Logger = NullLogger.Instance;
+        }
+
+        public ILogger Logger { get; set; }
+
+        public void OnAuthorization(AuthorizationContext filterContext) {
+
+            var accessToAuthorizedUserOnly = filterContext.ActionDescriptor.GetCustomAttributes(typeof(AuthorizeAttribute), true).Any() || filterContext.ActionDescriptor.ControllerDescriptor.ControllerType.GetCustomAttributes(typeof(AuthorizeAttribute), true).Any();
+
+            ////When user has logged out from a different browser, we have to invalidate all other browser sessions too.
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.Users/Services/SecurityFilter.cs]: Fix the comment style

**Comment by sebastienros** [comment]
Please fix the merge conflict

**Comment by BenedekFarkas** [comment]
@sebastienros the issue is marked for 1.10.x, but the PR targets dev, which one should we merge this into?

**Comment by BenedekFarkas** [comment]
@sebastienros I resolved the merge conflicts and tested the functionality. The CanAuthenticateWithCookie method will actually tell you if the User has logged out, because it's comparing LastLogoutUtc to LastLoginUtc, so it's actually working as expected and correctly logs the user out of the current browser if they logged out in another already (login is still working correctly). It looks g2g for me!

**Comment by sebastienros** [comment]
Where and when is the `LastLogoutUtc` stored? Does this filter imply a sql query for each http request?

File: src/Orchard.Web/Modules/Orchard.Users/Services/SecurityFilter.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+        public void OnAuthorization(AuthorizationContext filterContext) {
+            var accessToAuthorizedUserOnly =
+                filterContext.ActionDescriptor.GetCustomAttributes(typeof(AuthorizeAttribute), true).Any() ||
+                filterContext.ActionDescriptor.ControllerDescriptor.ControllerType.GetCustomAttributes(typeof(AuthorizeAttribute), true).Any();
+
+            // When user has logged out from a different browser, we have to invalidate all other browser sessions too.
+            // _membershipValidationService.CanAuthenticateWithCookie returns false if the user has logged out from a different browser.
+            if (accessToAuthorizedUserOnly && filterContext.RequestContext.HttpContext.Request.IsAuthenticated &&
+                !_membershipValidationService.CanAuthenticateWithCookie(_authenticationService.GetAuthenticatedUser())) {
```
Comment (MatteoPiovanelli) [inline src/Orchard.Web/Modules/Orchard.Users/Services/SecurityFilter.cs:28]: _membershipValidationService.CanAuthenticateWithCookie() is already called in _authenticationService.GetAuthenticatedUser() so we probably don't need it here as well

**Pull request closed without merging**